File: src/Questline.py
from Quest import Quest

class Questline:
    def __init__(self, name):
        self.quests = []
        self.name = name 
    
    def addQuest(self, desc):
       self.quests.append(Quest(desc, len(self.quests)))

    def __str__(self) -> str:
        s = self.name + "\n"
        for quest in self.quests:
            s += str(quest) + "\n"
        return s

File: src/questline.cpp
#include <string>
#include "questline.h"

using namespace std;

Questline::Questline(){}

Questline::Questline(string id, string name) {
    this->id = id; 
    this->name = name; 
}

void Questline::add(Quest q) {
    quests.push_back(q);
}
    
Quest Questline::get(int idx) {
    return quests[idx];
}

string Questline::getName() {
    return this->name; 
}